helper/red_black_tree.go

```diff
@@ -0,0 +1,90 @@
+package helper
+
+type Color bool
+
+const (
+	Red   Color = true
+	Black Color = false
+)
```

Comment:
Was a suggestion when strating the research. Additionally this could be extended to have the Children Property in the `RedBlackNode` and decide left or right by providing constants with the name